data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/configuration/KinesisSourceConfig.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+package org.opensearch.dataprepper.plugins.kinesis.source.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import lombok.Getter;
+import org.opensearch.dataprepper.model.configuration.PluginModel;
+
+import java.time.Duration;
+import java.util.List;
+
+public class KinesisSourceConfig {
+    static final Duration DEFAULT_TIME_OUT_IN_MILLIS = Duration.ofMillis(1000);
+    static final int DEFAULT_NUMBER_OF_RECORDS_TO_ACCUMULATE = 100;
+    static final Duration DEFAULT_SHARD_ACKNOWLEDGEMENT_TIMEOUT = Duration.ofMinutes(10);
+
+    @Getter
+    @JsonProperty("streams")
+    @NotNull
+    @Valid
+    @Size(min = 1, max = 4, message = "Provide 1-4 streams to read from.")
+    private List<KinesisStreamConfig> streams;
+
+    @Getter
+    @JsonProperty("aws")
+    @NotNull
+    @Valid
+    private AwsAuthenticationConfig awsAuthenticationConfig;
+
+    @Getter
+    @JsonProperty("buffer_timeout")
+    private Duration bufferTimeout = DEFAULT_TIME_OUT_IN_MILLIS;
+
+    @Getter
+    @JsonProperty("records_to_accumulate")
+    private int numberOfRecordsToAccumulate = DEFAULT_NUMBER_OF_RECORDS_TO_ACCUMULATE;
+
+    @JsonProperty("acknowledgments")
+    @Getter
+    private boolean acknowledgments = false;
```

Comment:
Should we start making acknowledgements as `true` by default to all pull sources? If yes, we should start with this. @dlvenable

Reply:
I think we should keep the behavior consistent for now. I do think we should revisit in Data Prepper 3.0. It may make sense to have a `disable_acknowledgements: true` in that case. But, let's do it consistently then.